relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,108 @@
+use std::borrow::Cow;
+use std::time::Duration;
+
+use relay_dynamic_config::GlobalConfig;
+use relay_system::{Addr, Service};
+use reqwest::Method;
+use serde::{Deserialize, Serialize};
+
+use crate::actors::processor::EnvelopeProcessor;
+use crate::actors::upstream::{RequestPriority, SendQuery, UpstreamQuery, UpstreamRelay};
+
+/// Service implementing the [`GlobalConfig`] interface.
+///
+/// The service is responsible for fetching the global config and
+/// forwarding it to the services that require it.
+#[derive(Debug)]
+pub struct GlobalConfigService {
+    envelope_processor: Addr<EnvelopeProcessor>,
+    upstream: Addr<UpstreamRelay>,
+}
+
+impl GlobalConfigService {
+    pub fn new(envelope_processor: Addr<EnvelopeProcessor>, upstream: Addr<UpstreamRelay>) -> Self {
+        Self {
+            envelope_processor,
+            upstream,
+        }
+    }
+
+    /// Forwards the given global config to the services that require it.
+    fn update_global_config(&mut self) {
+        let upstream_relay: Addr<UpstreamRelay> = self.upstream.clone();
+        let envelope_processor = self.envelope_processor.clone();
+        tokio::spawn(async move {
+            let query = GetGlobalConfig {
+                global_config: true,
+            };
+
+            match upstream_relay.send(SendQuery(query)).await {
+                Ok(Ok(response)) => {
+                    envelope_processor.send::<GlobalConfig>(response.global);
+                }
+                Err(e) => {
+                    relay_log::error!("failed to send request: {}", e);
+                }
+                Ok(Err(e)) => {
+                    relay_log::error!("failed to fetch request: {}", e);
+                }
+            };
+        });
+    }
+}
+
+impl Service for GlobalConfigService {
+    type Interface = ();
+
+    fn spawn_handler(mut self, _rx: relay_system::Receiver<Self::Interface>) {
+        tokio::spawn(async move {
+            let mut ticker = tokio::time::interval(Duration::from_secs(10));
+            relay_log::info!("global config started");
+
+            loop {
+                tokio::select! {
+                    biased;
+                    _ = ticker.tick() => self.update_global_config(),
+                    else => break,
+                }
+            }
+            relay_log::info!("global config stopped");
+        });
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfig {
+    global_config: bool,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfigResponse {
+    global: GlobalConfig,
```

Comment:
Let's make this an `Arc` from the start. This will allow to send the same config to multiple services without deep copying, and makes later `Arc`ing in the `EnvelopeProcessor` unnecessary.

```suggestion
    global: Arc<GlobalConfig>,
```

Reply:
will it implicitly turn it into an Arc here?

```Rust
            match upstream_relay.send(SendQuery(query)).await {
                Ok(Ok(response)) => {
                    envelope_processor.send::<Arc<GlobalConfig>>(response.global);
                }
```